gradle/libs.versions.toml

```diff
@@ -11,11 +11,9 @@ ktlint = "0.50.0"
 moshi = "1.15.0"
 
 minSdk = "25"
-compileSdk = "33"
+compileSdk = "34"
 
-# Maps to this commit: https://android.googlesource.com/platform/prebuilts/studio/layoutlib/+/5e972ea
-layoutlib = "2022.3.1-5e972ea"
-layoutlibPrebuiltSha = "5e972ea"
+layoutlib = "14.0.1"
```

Comment:
How much of this PR is actually part of #1216 and #1220?

Reply:
The compileSdkVersion bump can and probably needs to already happen with the Iguana update. As for the API changes, I'd need to try building it to find out.

Reply:
I've rebased this PR and restructured the commits. The first commit bumps the version to Iguana 2023.2.1-beta02-6c7316c, the following commits then apply the required changes for Iguana and the last commit bumps to Jellyfish and Google's artifacts. All the changes here except for that last commit are already needed to make the Iguana layoutlib run.

Iguana has the same issue with the 3 tests that are failing on either macOS or on Windows+Linux, depending on where the snapshot was created. I made a test run with a manually published 2023.2.1-beta02-6c7316c here https://github.com/freeletics/paparazzi/actions/runs/7590152036